integration/init_test.go

```diff
@@ -98,11 +103,17 @@ func TestIntegration(t *testing.T) {
 		Execute(settings.Config.BuildPlan)
 	Expect(err).NotTo(HaveOccurred())
 
+	pack := occam.NewPack().WithVerbose()
+	builder, err = pack.Builder.Inspect.Execute()
+	Expect(err).NotTo(HaveOccurred())
+
 	SetDefaultEventuallyTimeout(10 * time.Second)
 
 	suite := spec.New("Integration", spec.Report(report.Terminal{}))
 	suite("Default", testDefault, spec.Parallel())
 	suite("LayerReuse", testLayerReuse, spec.Parallel())
-	suite("Offline", testOffline, spec.Parallel())
+	if strings.Contains(builder.LocalInfo.Stack.ID, "jammy") || strings.Contains(builder.LocalInfo.Stack.ID, "bionic") {
```

Comment:
@jpena-r7 I'm curious if you actually need this conditional. Given you haven't made any changes to the "compilation" process, I would expect the offline tests to pass on all stacks. Did you already try it? Did it fail for a reason I'm not seeing?

Even if the tests pass and we can remove the conditional, I could see the argument for consistency across the rest of the buildpacks (i.e. cpython cannot run offline tests with arbitrary stacks). But i think the value in running all the tests against all stacks - and having less complexity in the tests by removing conditionals - outweighs that.

Reply:
I did try it without the conditional, but it failed because it couldn't find the python3 executable when trying adding the symlink (see below). This makes sense since if the offiline tests only deliver the dependency (python source), which means the executables would be missing since they haven't been compiled yet.

```
=== CONT  TestIntegration/Integration/Offline/when_the_buildpack_is_run_with_pack_build_in_offline_mode/builds_successfully
    offline_test.go:66: 
        ===> ANALYZING
....
        [builder] symlink /layers/paketo-buildpacks_cpython/cpython/bin/python3 /layers/paketo-buildpacks_cpython/cpython/bin/python: no such file or directory
        [builder] ERROR: failed to build: exit status 1
        ERROR: failed to build: executing lifecycle. This may be the result of using an untrusted builder: failed with status code: 51
```

From my perspective, offline tests should work for cpython (and dependent buildpacks like pip) because you should be able to compile from source once the dependency has been added to the freezer cache, but it doesn't look like build function gets run when it is marked as offline.

I agree completely that tests should not be excluded, but I suspect this will require a change to how the offline test works. I am not familiar enough with occam to know how easy this would be, but happy to take a look with a little guidance.

Reply:
Ah, ok, so it's a real test failure then. Let's leave the conditional in.

As far as the offline behavior - I'd have to dig into this some more. I think we could consider "offline support for arbitrary stacks" as a separate track that would encompass `cpython`, `pip`, and `pip-install` as necessary.